app/screens/PublicLinkScreen/PublicLinkScreen.tsx

```diff
@@ -49,6 +52,45 @@ export default function PublicLinkScreen ({ navigation, route }: PublicLinkScree
     });
   }
 
+
+  const handleShareAsPdf = async() => {
+    // templateURL = svg template from id of renderMethod in Credential
+    const templateURL = credential.renderMethod?.[0].id; // might want to sort if there are more than one renderMethod
+    let source = '';
+    if (templateURL) {
+      await fetch(templateURL)
+        .then(response => {
+          return response.text();
+        }).then(data => {
+          source = data;
+        }).catch(error => {
+          console.error('An error occurred fetching the SVG Template:', error);
+        });
```

Comment:
Ok, so, `fetch()` is treacherous. You'd THINK that if there's an HTTP error (like 404 Not Found), the `.catch()` at the end would catch it. But not so :) You have to mess with `response.ok` instead. 
So the correct version of this line is something like:

```suggestion
      let source;
      try {
        const response = await fetch(templateURL);
        if (!response.ok) {
          throw new Error(`HTTP Error: ${response.status}`);
        }
        source = await response.json();
      } catch (e) {
        console.log('Error fetching template:', e);
      }
```